plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/navigation/openimplementation/OpenImplementationPresenter.java

```diff
@@ -96,161 +91,149 @@ public void show(final EditorPartPresenter editorPartPresenter) {
+    //          .then(
+    //              new Operation<JarEntry>() {
+    //                @Override
+    //                public void apply(final JarEntry entry) throws OperationException {
+    //                  service
+    //                      .getContent(
+    //                          project.get().getLocation(),
+    //                          member.getLibId(),
+    //                          Path.valueOf(entry.getPath()))
+    //                      .then(
+    //                          new Operation<ClassContent>() {
+    //                            @Override
+    //                            public void apply(ClassContent content) throws OperationException
+    // {
+    //                              final String clazz =
+    //                                  entry.getName().substring(0, entry.getName().indexOf('.'));
+    //                              final VirtualFile file =
+    //                                  new SyntheticFile(entry.getName(), clazz,
+    // content.getContent());
+    //                              editorAgent.openEditor(
+    //                                  file,
+    //                                  new OpenEditorCallbackImpl() {
+    //                                    @Override
+    //                                    public void onEditorOpened(EditorPartPresenter editor) {
+    //                                      setCursor(member.getFileRegion());
+    //                                    }
+    //                                  });
+    //                            }
+    //                          });
+    //                }
+    //              });
+    //    } else {
+    context
+        .getWorkspaceRoot()
+        .getFile(symbolInformation.getLocation().getUri())
+        .then(
+            file -> {
+              if (file.isPresent()) {
+                editorAgent.openEditor(
+                    file.get(),
+                    new OpenEditorCallbackImpl() {
+                      @Override
+                      public void onEditorOpened(EditorPartPresenter editor) {
+                        setCursor(symbolInformation.getLocation().getRange());
+                      }
+                    });
               }
             });
+    // }
+    Scheduler.get().scheduleDeferred(() -> activeEditor.setFocus());
   }
 
   private void showNoImplementations(
       NoImplementationWidget noImplementationWidget, TextEditor editorPartPresenter) {
     int offset = editorPartPresenter.getCursorOffset();
     PositionConverter.PixelCoordinates coordinates =
         editorPartPresenter.getPositionConverter().offsetToPixel(offset);
-    Type type = dtoFactory.createDto(Type.class);
-    type.setFlags(-1);
-    noImplementationWidget.addItem(type);
+    SymbolInformation symbolInformation = new SymbolInformation();
+    symbolInformation.setKind(null);
+    noImplementationWidget.addItem(symbolInformation);
```

Comment:
PopupWidget handles the case of no elements. Why do we complicate our lives here?

Reply:
PopupWidget seems to show a settable text when no elements are present (sth. like "No implementations"), so no need to invent an artificial element.